lib/puppet/provider/service/smf.rb

```diff
@@ -129,6 +129,14 @@ def status
       return
     end
 
+    # Check to see if the service exists
+    cmd = Array[command(:svccfg), "select", @resource[:name]]
+    output = Puppet::Util::Execution.execute(
+        cmd, :combine => true, :failonfail => false)
+    if $CHILD_STATUS.exitstatus != 0
+      info output
+    end
```

Comment:
@shawnferry while reviewing this part I realized it appears as if this provider doesn't distinguish between a non-existent service from a stopped/disabled and absent services and can't ensure => absent. I.e., `foo` does not actually exist:
```
-bash-4.1# puppet resource service foo
service { 'foo':
  ensure => 'stopped',
  enable => 'false',
}
```

I filed [PUP-6860](https://tickets.puppetlabs.com/browse/PUP-6860) to address this.

Separate from that, can you pull this existence check logic out into its own method in the provider? I'd like to fix the aforementioned bug and anticipate being able to use it if you could.

Second, I think the way we try to do this now is to reference the command as a method, and use Puppet's logging methods rather than the logging functions (though I admit I'm not 100% sure on that last one). ie:
```
begin
  svccfg(:select, @resource[:name])
rescue Puppet::ExecutionFailure => detail
  Puppet.debug("service was not found, svccfg returned: #{detail.inspect}")
  return false
end
true
```